Public/Src/Tools/Tool.MsBuildGraphBuilder/PropertyTrackingLogger.cs

```diff
@@ -0,0 +1,77 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using BuildXL.Utilities.Collections;
+using Microsoft.Build.Framework;
+
+namespace ProjectGraphBuilder
+{
+    internal class PropertyTrackingLogger : ILogger
+    {
+        private readonly HashSet<string> _variablesRead = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+        private bool _nonTrackingSdkResolversExist = false;
+
+        public IEnumerable<string> PotentialEnvironmentVariablesReads

```

Comment:
>PotentialEnvironmentVariablesReads [](start = 35, length = 34)

Can we add a contract here so this property is only queried when tracking is available?

Reply:
This property was changed. See the new logic in upcoming commit.